src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -382,37 +384,30 @@ public static byte[] UrlEncodeToBytes(byte[] value, int offset, int count)
                 return null;
             }
 
-            int cSpaces = 0;
-            int cUnsafe = 0;
+            bool foundSpaces = false;
+            int unsafeCount = 0;
 
             // count them first
             for (int i = 0; i < count; i++)
             {
                 char ch = (char)value[offset + i];
 
                 if (ch == ' ')
-                    cSpaces++;
+                    foundSpaces = true;
                 else if (!IsUrlSafeChar(ch))
-                    cUnsafe++;
+                    unsafeCount++;
             }
 
             // nothing to expand?
-            if (cSpaces == 0 && cUnsafe == 0)
+            if (!foundSpaces && unsafeCount == 0)
```

Comment:
Interesting.
R.e. stackalloc: that's a pretty good idea and has proven allocation and performance benefits.

I'm wondering, is there interest in an API that lets you allocate a char* or byte* etc. variably as a fixed char[] or a stackalloced char*

E.g. have a convenient way to remove this boilerplate:
```
int count = ...; // Any value
int stackallocThreshold = 512;
if (count < stackallocThreshold)
{
    char* pChars = stackalloc char[count];
    // Do something with pChars    
}
else
{
    char[] chars = new char[count];
    fixed (char* pChars = chars)
    {
        // Do something with pChars
    }
}
```